src/core/service.c

```diff
@@ -1512,9 +1512,9 @@ static int service_allocate_exec_fd_event_source(
         if (r < 0)
                 return log_unit_error_errno(UNIT(s), r, "Failed to allocate exec_fd event source: %m");
 
-        /* This is a bit lower priority than SIGCHLD, as that carries a lot more interesting failure information */
+        /* This is a bit higher priority than SIGCHLD to avoid a race. See #28304 */
```

Comment:
please explain the race briefly here. i.e. something like "This is a bit higher priority than SIGCHLD to make sure we do not mistake a process starting successfully and quickly failing after that with a process failing to start."